backend/database.js

```diff
@@ -184,15 +185,31 @@ function throwIfFileOutsideDataDir (filename: string): string {
 }
 
 if (production || process.env.GI_PERSIST) {
+  // https://github.com/isaacs/node-lru-cache#usage
+  const cache = new LRU({
+    max: 1000
```

Comment:
Can you make this 10k and add a `// TODO:` comment to read this value from a file / envvar?

In fact, you can also implement the envar now, e.g. with `process.env.GI_LRU_NUM_ITEMS`. Just remember envars need to be converted from strings to numbers first.